go.mod

```diff
@@ -123,22 +97,22 @@ require (
 	github.com/weaveworks/mesh v0.0.0-20170419100114-1f158d31de55
 	github.com/zclconf/go-cty v1.3.1
 	go.uber.org/zap v1.10.0
-	golang.org/x/crypto v0.0.0-20191202143827-86a70503ff7e
+	golang.org/x/crypto v0.0.0-20200220183623-bac4c82f6975
 	golang.org/x/net v0.0.0-20200202094626-16171245cfb2
 	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
 	golang.org/x/sys v0.0.0-20191128015809-6d18c012aee9
 	golang.org/x/tools v0.0.0-20191203134012-c197fd4bf371
 	google.golang.org/api v0.17.0
 	gopkg.in/gcfg.v1 v1.2.3
 	gopkg.in/inf.v0 v0.9.1
-	gopkg.in/yaml.v2 v2.2.7
+	gopkg.in/yaml.v2 v2.2.8
 	honnef.co/go/tools v0.0.1-2019.2.3
-	k8s.io/api v0.0.0
-	k8s.io/apimachinery v0.0.0
-	k8s.io/cli-runtime v0.0.0
+	k8s.io/api v0.17.4
+	k8s.io/apimachinery v0.17.4
+	k8s.io/cli-runtime v0.17.4
 	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
```

Comment:
thanks, the video was very helpful! can you explain why this version didn't (need to) change? It seems all of the k8s.io packages are covered by `replace`s above, but this, kubectl, and legacy-cloud-providers below didn't change.

Reply:
Hmmm ... that's a good point and is just my mistake!  Will hold the PR!

Reply:
Ah actually, I take it back...

The replace directives at the top take precedence, and we do have:

```
replace k8s.io/client-go => k8s.io/client-go v0.17.4
```

client-go started adopted a different versioning scheme though, and it looks like it isn't fully sorted out yet.  That confuses go.mod which then puts some random version in the directives.

I was able to override the version in go.mod manually, so it now looks intuitive, even though that version wasn't used before.

We'll likely have to do this manually going forwards (but if we forget, it shouldn't really matter)

Reply:
ah ok. and does the same apply for kubectl and legacy-cloud-providers ?